src/neynarAPI/NeynarAPIClient.ts

```diff
@@ -0,0 +1,679 @@
+      if (response.data.result?.next?.cursor === undefined) {
+        break;
+      }
+      cursor = response.data.result.next.cursor ?? undefined;
+    }
+  }
+
+  /**
+   * Fetch all likes by a given user.
+   */
+  public async *fetchUserCastLikes(
+    user: { fid: number },
+    { pageSize = 100 } = {}
+  ): AsyncGenerator<ReactionWithCastMeta, void, undefined> {
+    let cursor: string | undefined;
+    let viewer: number | undefined;
+
+    while (true) {
+      // fetch one page of likes
+      const response = await this.apis.user.userCastLikes(
+        user.fid,
+        viewer,
+        pageSize,
+        cursor
+      );
+
+      // Extract the likes array from the response (or provide an empty array as a default)
+      const likes =
+        response.data.result?.likes != null ? response.data.result.likes : [];
+
+      // Create an async iterable for the likes
+      for (const like of likes) {
+        yield like;
+      }
+
+      // prep for next page
+      if (response.data.result?.next?.cursor === undefined) {
+        break;
+      }
+      cursor = response.data.result.next.cursor ?? undefined;
+    }
+  }
+
+  /**
+   * Gets the specified user via their FID (if found)
+   */
+  public async lookupUserByFid(fid: number): Promise<User | undefined> {
+    try {
+      const response = await this.apis.user.user(fid);
+      return response.data.result?.user;
+    } catch (error) {
+      if (NeynarAPIClient.isApiErrorResponse(error)) {
+        if (error.response.status === 404) return undefined;
+      }
+      throw error;
+    }
+  }
+
+  /**
+   * Gets the specified user via their username (if found)
+   */
+  public async lookupUserByUsername(
+    username: string
+  ): Promise<User | undefined> {
+    let viewer: number | undefined;
+    const response = await this.apis.user.userByUsername(username, viewer, {
+      validateStatus: (status: number) => {
+        return status === 200 || status === 404;
+      },
+    });
+    if (response.status === 404) return undefined;
+    return response.data.result?.user;
+  }
+
+  /**
+   * Gets the custody address for the specified user via their username (if found)
+   */
+  public async fetchCustodyAddressForUser(
+    fid: number
+  ): Promise<string | undefined> {
+    const response = await this.apis.user.custodyAddress(fid);
+    return response.data.result.custodyAddress ?? undefined;
+  }
+
+  public async fetchUserVerifications(user: {
+    fid: number;
+  }): Promise<VerificationResponseResult | undefined> {
+    const response = await this.apis.verification.verifications(user.fid);
+    return response.data.result;
+  }
+
+  /**
+   * Checks if a given Ethereum address has a Farcaster user associated with it.
+   * Note: if an address is associated with multiple users, the API will return
+   * the user who most recently published a verification with the address
+   * (based on when Merkle received the proof, not a self-reported timestamp).
```

Comment:
- Is this still true?
- If so, need to fix reference to 'Merkle'

Reply:
I'm not sure, I'll need to test this

Reply:
I was unable to verify this behavior but this comment is taken from their [docs](https://docs.neynar.com/reference/get-user-by-verification)